src/averageAges.js

```diff
@@ -1,61 +1,60 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
-function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+function calculateMenAverageAge(people, century = people) {
+  let calculateCentury = century;
+
+  if (Number(century)) {
+    calculateCentury = people
+      .filter(peopleCentury => Math.ceil(peopleCentury.died / 100) === century);
+  }
+
+  const mans = calculateCentury
+    .filter(peopleSex => peopleSex.sex === 'm');
+  const calculateManAverageAge = mans
+    .reduce((sum, a) => sum + (a.died - a.born), 0) / mans.length;
+
+  return calculateManAverageAge;
 }
 
-/**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
- *
- * @param {object[]} people
- * @param {boolean} withChildren - optional
- *
- * @return {number}
- */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const woman = people
+    .filter(peopleSex => peopleSex.sex === 'f');
+
+  const calculateWomanAverageAge = woman
+    .reduce((sum, a) => sum + (a.died - a.born), 0) / woman.length;
+
+  const withChild = people
+    .filter(name => people.find(child => child.mother === name.name));
+
+  const calculateAverageAgeWithChild = withChild
+    .reduce((sum, a) => sum + (a.died - a.born), 0) / withChild.length;
```

Comment:
you repeat logic for calculation basically for the same array, don't do that, instead filter array either for `withChildren` or without it and perform calculations on newly created array